src/kinodynamic-path.cc

```diff
@@ -0,0 +1,307 @@
+            if(tv_[id] > 0 )
+              v2 = vLim_[id];
+            else
+              v2 = t1_[id]*a1_[id] + initial_[indexVel];
+            result[id] = 0.5*t1_[id]*t1_[id]*a1_[id] + t1_[id]*initial_[indexVel] + initial_[id] + t0_[id]*initial_[indexVel] + tv_[id]*vLim_[id] - 0.5*t2*t2*a1_[id] + t2*v2;
+            result[indexVel] = v2 - t2 * a1_[id];
+            result[indexAcc] = -a1_[id];
+
+          }else{ // after last segment : constant velocity (null ?)
+            if(end_[indexVel] != 0 && (t - (t0_[id] + t1_[id] + tv_[id] +t2_[id])) > 0.00015 ){ // epsilon from rbprm-steering-kinodynamic
+              hppDout(notice,"Should not happen : You should not have a constant velocity segment at the end if the velocity is not null");
+              hppDout(notice,"index Joint = "<<id);
+              hppDout(notice,"t = "<<t<<"  ;  length = "<<t0_[id] + t1_[id] + tv_[id] +t2_[id]);
+            }
+            result[id] = end_[id];
+            result[indexVel] = end_[indexVel];
+            result[indexAcc] = 0;
+          }
+        }// if not quaternion joint
+
+     // }// if joint config size > 1
+      
+     }// for all joints
+      
+      return true;
+    }
+    
+    PathPtr_t KinodynamicPath::extract (const interval_t& subInterval) const
+    throw (projection_error)
+    {
+      assert(subInterval.first >=0 && subInterval.second >=0 && "Negative time values in extract path");
+      assert(subInterval.first >=timeRange_.first && subInterval.second <= timeRange_.second && "Given interval not inside path interval");
+      // Length is assumed to be proportional to interval range
+      if(subInterval.first == timeRange().first && subInterval.second == timeRange().second){
+        hppDout(notice,"Call extract with same interval");
+        return weak_.lock();
+      }
+      value_type l = fabs (subInterval.second - subInterval.first);
+      if(l<=0){
+        hppDout(notice,"Call Extract with a length null");
+        return PathPtr_t();
+      }
+
+      hppDout(notice,"%% EXTRACT PATH : path interval : "<<timeRange_.first<<" ; "<<timeRange_.second);
+      hppDout(notice,"%% EXTRACT PATH : sub  interval : "<<subInterval.first<<" ; "<<subInterval.second);
+
+      bool success;
+      Configuration_t q1 ((*this) (subInterval.first, success));
+      if (!success) throw projection_error
+          ("Failed to apply constraints in KinodynamicPath::extract");
+      Configuration_t q2 ((*this) (subInterval.second, success));
+      if (!success) throw projection_error
+          ("Failed to apply constraints in KinodynamicPath::extract");
+
+      // set acceleration to 0 for initial and end config :
+      size_type configSize = device()->configSize() - device()->extraConfigSpace().dimension ();
+      q1[configSize+3] = 0.0;
+      q1[configSize+4] = 0.0;
+      q1[configSize+5] = 0.0;
+      q2[configSize+3] = 0.0;
+      q2[configSize+4] = 0.0;
+      q2[configSize+5] = 0.0;
```

Comment:
Would you mind adding to the documentation of the class that it expects an extra config space whose 6 first dimensions are the acceleration of the base ?

Reply:
Done in https://github.com/humanoid-path-planner/hpp-core/pull/117/commits/fa5678a3e03019b2cf22b2c0a905189272125848